### PR TITLE
Feat: Add debugging logs control

### LINKED ISSUE #216: Make debugging easier by including query in postgres error
More often than not I have to log my executed queries to find syntax problems but have to do it outside of the query in order to get the executed query

### PR DESCRIPTION
**Pull request by bombillazo**
Fixes #216, #254

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -156,6 +153,19 @@ This situation will stabilize as `std` and `deno-postgres` approach version 1.0.
 | 1.17.0        | 0.15.0             | 0.17.1             |                      |
 | 1.40.0        | 0.17.2             |                    | Now available on JSR |
 
+## Breaking changes
+
+Although the `deno-driver` is pretty stable and robust, it is a WIP and we're
+still exploring the design. Expect some breaking changes as we reach version 1.0
+and enhance the feature set. PLease check the Releases for more info on breaking
```
Comment (iuioiua) [inline README.md:160]:
```suggestion
and enhance the feature set. Please check the Releases for more info on breaking
```
It'd be good to have the `Releases` be a link.

File: README.md [hunk 2]
```diff
@@ -156,6 +153,19 @@ This situation will stabilize as `std` and `deno-postgres` approach version 1.0.
...
+changes. Please reach out if there are any undocumented breaking changes.
+
+## Found issues?
+
+Please file an issue with any problems with the driver in this repository's
```
Comment (iuioiua) [inline README.md:165]:
```suggestion
Please [file an issue](https://github.com/denodrivers/postgres/issues/new/choose) with any problems with the driver in this repository's
```

File: README.md [hunk 3]
```diff
@@ -156,6 +153,19 @@ This situation will stabilize as `std` and `deno-postgres` approach version 1.0.
...
+issue section. If you would like to help, please look at the issues as well. You
```
Comment (iuioiua) [inline README.md:166]:
```suggestion
issue section. If you would like to help, please look at the [issues](https://github.com/denodrivers/postgres/issues) as well. You
```

File: README.md [hunk 4]
```diff
@@ -156,6 +153,19 @@ This situation will stabilize as `std` and `deno-postgres` approach version 1.0.
 | 1.17.0        | 0.15.0             | 0.17.1             |                      |
 | 1.40.0        | 0.17.2             |                    | Now available on JSR |
 
+## Breaking changes
+
+Although the `deno-driver` is pretty stable and robust, it is a WIP and we're
```
Comment (iuioiua) [inline README.md:158]:
```suggestion
Although `deno-postgresql` is pretty stable and robust, it is a WIP, and we're
```

File: debug.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Controls debugging behavior {@default false}
+ *
+ * - `true` : all debug options are enabled
+ * - `false` : all debug options are disabled
+ * - DebugOptions:
+ *   - `query` : Log queries
+ *   - `notices` : Log notices
```
Comment (iuioiua) [inline debug.ts:8]:
```suggestion
 * Controls debugging behavior. If set to `true`, all debug options are enabled.
 * If set to `false`, all debug options are disabled.
 *
 * {@default false}
```
I don't think we need to repeat information for `DebugOptions`.

File: docs/README.md [hunk 6]
```diff
@@ -1393,3 +1393,37 @@ await transaction.queryArray`INSERT INTO DONT_DELETE_ME VALUES (2)`; // Still in
...
+- `notices` : Logs database notices
+- `results` : Logs the result of the queries
+
+```ts
+{
+  const client = new Client({
+    ...
+    controls: {
+      // enable all logs
+      debug: true,
```
Comment (iuioiua) [inline docs/README.md:1415]: Might it be worth just having `debug` be a boolean? Just an idea.

**Review by iuioiua — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 6 above)
Comment (bombillazo) [reply docs/README.md:1415]: I'd let developers choose the debugging options they need. I've had cases where enabling all debugging options is simply too noisy to be useful.

**Pull request merged**